rules/Orm30/Rector/MethodCall/SetParametersArrayToCollectionRector.php

```diff
@@ -0,0 +1,209 @@
+                    new \Doctrine\ORM\Query\Parameter('foo', 'bar')
+                ]));
+                CODE_SAMPLE
+                )]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Node\Stmt\ClassMethod::class, MethodCall::class];
+    }
+
+    /**
+     * @param Node\Stmt\ClassMethod|MethodCall $node
+     */
+    public function refactor(Node $node): Node|null
+    {
+        if ($node instanceof Node\Stmt\ClassMethod) {
+            $variables = $this->getAffectedVariables($node);
+            foreach ($variables as $variable) {
+                $this->iterateThroughStmts($node->getStmts(), $variable);
+            }
+
+            return $node;
+        }
+
+        return $this->refactorMethodCall($node);
+    }
+
+    /**
+     * @param list<Node\Stmt> $stmts
+     */
+    public function iterateThroughStmts(array $stmts, Node\Expr\Variable $variable): void
+    {
+        foreach ($stmts as $stmt) {
+            if ($stmt instanceof StmtsAwareInterface) {
+                $this->iterateThroughStmts($stmt->stmts, $variable);
+                continue;
+            }
+
+            if (
+                $stmt->expr instanceof Node\Expr\Assign
+                && $stmt->expr->expr instanceof Array_
+                && $stmt->expr->var->name === $variable->name
+            ) {
+                $newCollection = new New_(new FullyQualified('Doctrine\\Common\\Collections\\ArrayCollection'));
+                $newCollection->args = [new Arg(new Array_($this->convertArrayToParameters($stmt->expr->expr)))];
+                $stmt->expr->expr = $newCollection;
+            }
+
+            if (
+                $stmt->expr instanceof Node\Expr\Assign
+                && $stmt->expr->var instanceof Node\Expr\ArrayDimFetch
+                && $stmt->expr->var->var->name === $variable->name
+            ) {
+                $newParameter = new New_(new FullyQualified('Doctrine\\ORM\\Query\\Parameter'));
+                $newParameter->args = [new Arg($stmt->expr->var->dim), new Arg($stmt->expr->expr)];
+
+                $stmt->expr = new MethodCall($stmt->expr->var->var, 'add', [new Arg($newParameter)]);
+            }
+        }
+    }
+
+    /**
+     * @return list<Node\Expr\Variable>
+     */
+    private function getAffectedVariables(Node\Stmt\ClassMethod $classMethod): iterable
+    {
+        foreach ($classMethod->getStmts() as $stmt) {
+            if (
+                $stmt instanceof Node\Stmt\Expression
+                && $stmt->expr instanceof MethodCall
+                && $stmt->expr->args[0]?->value instanceof Node\Expr\Variable
+                && $stmt->expr->name->name === 'setParameters'
+            ) {
+                $varType = $this->nodeTypeResolver->getType($stmt->expr->var);
+                if (! $varType->isInstanceOf('Doctrine\\ORM\\QueryBuilder')->yes()) {
```

Comment:
Instead of `not yes` you could just use `no` so

```php
! $varType->isInstanceOf('Doctrine\\ORM\\QueryBuilder')->yes()
```
can be changed to:

```php
$varType->isInstanceOf('Doctrine\\ORM\\QueryBuilder')->no()
```

Reply:
overall i would say the same but the varType as also the type `MAYBE` - And in that cases i do not want to modify the values. But i will try to test this also with an fixture